Support arrow input.

evetion opened the pull request:
Fixes #170

What do we support in terms of input/relative dir as put in the settings? Currently, this ignores the setting completely.

visr commented:
Ah, great, surprised that this just works on the Julia side. Indeed since this is input I'd say the path has to be relative to both the TOML directory and the optional `input_dir`.

Actually, I see `input_dir` and `results_dir` are missing from `docs.toml`, if you feel like addressing that as well.

evetion commented:
> Ah, great, surprised that this just works on the Julia side. Indeed since this is input I'd say the path has to be relative to both the TOML directory and the optional `input_dir`.
> 
> Actually, I see `input_dir` and `results_dir` are missing from `docs.toml`, if you feel like addressing that as well.

I can add the input_dir, and leave the relative_dir for now, as I think it's unused and should probably be removed. Other naming things should probably be refined and changed in another PR.

visr commented:
Great, thanks. I thought `relative_dir` isn't really a TOML key, it is just a way on the Julia side to interpret all paths relative to the TOML.

evetion commented:
Ok, I've now added using input_dir in both reading and writing. I've also added a test in Python to check we can read arrow files back in again (from the relative input_path).

Hofer-Julian commented on docs/python/examples.ipynb:
```diff
@@ -471,7 +471,7 @@
    "outputs": [],
    "source": [
-    "model = ribasim.Model.from_toml(datadir / \"basic/ribasim.toml\")"
+    "model = ribasim.Model(filepath=datadir / \"basic/ribasim.toml\")"
```
I liked having separate constructors here. Is there anything blocking us from being explicit here?
Also, what is the behavior if I now specify `filepath` as well as other parameters? Will the TOML or the manually set parameters take precedence?

evetion replied:
Hmm, I found it confusing. Would `read` also be acceptable (as there's a `write`)?

Manually set should override, I've fixed that now.

Hofer-Julian replied:
`read` sounds good. There is no alternative to our toml.

evetion commented:
Thanks for the review, all addressed now.

Hofer-Julian commented on core/test/docs.toml:
```diff
@@ -37,9 +39,9 @@ timing = false     # optional, whether to log debug timing statements
-flow = "results/flow.arrow"       # optional, default "results/flow.arrow"
-basin = "results/basin.arrow"     # optional, default "results/basin.arrow"
-control = "results/control.arrow" # optional, default "results/control.arrow"
+flow = "results/flow.arrow"             # optional, default "results/flow.arrow"
```
Why do you still have to specify "results" here? :)

Shouldn't this just be flow.arrow now?

evetion replied:
Yeah probably, but it might be good to have that discussion first before changing it. Results are out of scope for this PR?

Hofer-Julian replied:
But you introduced results_dir.

Shouldn't this part be adapted accordingly?

Or am I misunderstanding what results_dir does?

evetion replied:
This PR does not introduce results_dir functionality. I added it to the docs.toml because it was missing there on request of @visr, and autoformatting did a bit more to that file. The Python code does not (yet) read results files.

Hofer-Julian replied:
I see.
If this is unrelated to your PR then let's merge it.
Let's remember to discuss that result_dir thing on Monday :)
